src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -42,9 +43,11 @@ internal sealed class TensorFlowMapper : IRowMapper
             private readonly TFShape[] _tfInputShapes;
             private readonly TFDataType[] _tfInputTypes;
 
-            private readonly string _outputColName;
-            private readonly ColumnType _outputColType;
-            private readonly TFDataType _tfOutputType;
+            private readonly string[] _outputColNames;
+            private readonly ColumnType[] _outputColTypes;
+            private readonly TFDataType[] _tfOutputTypes;
+            private IDictionary<string, TFTensor> _cachedOutputs;
```

Comment:
@yaeldekel, is it the right place to cache data here?

Reply:
The dictionary holding the cached outputs needs to be instantiated on each call to CreateGetters.

---
In reply to: [215089636](https://github.com/dotnet/machinelearning/pull/814#discussion_r215089636) [](ancestors = 215089636)

Reply:
cachedPosition as well.

---
In reply to: [215387466](https://github.com/dotnet/machinelearning/pull/814#discussion_r215387466) [](ancestors = 215387466,215089636)